hawkular-inventory-api/src/main/java/org/hawkular/inventory/api/Inventory.java

```diff
@@ -101,6 +110,91 @@
  * @since 0.0.1
  */
 public interface Inventory extends AutoCloseable, Tenants.Container<Tenants.ReadWrite> {
+    static final class InventotyUtils {
+        private static final Map<String, Class<? extends Entity<?, ?>>>entityTypes;
+        static {
+            Map<String, Class<? extends Entity<?, ?>>> m = Arrays.asList(
+                    Tenant.class,
+                    Environment.class,
+                    Feed.class,
+                    ResourceType.class,
+                    MetricType.class,
+                    Resource.class,
+                    Metric.class,
+                    OperationType.class,
+                    MetadataPack.class)
+            .stream().collect(Collectors.toMap(Class::getName, Function.identity()));
+            entityTypes = Collections.unmodifiableMap(m);
+        }
+        static String shortenIfNeeded(CanonicalPath path) {
```

Comment:
I commented on that line. IMHO, it has been incorrect the whole time and should be replaced with CanonicalPath.toString().